api/routes/api_news.py

```diff
@@ -79,8 +80,19 @@ def specific_article(service: NewsService = Provide[Container.news_service]):
         logger.error(f"Error in POST /: {str(e)}")
         get_custom_error_response(e)
 
+@news_app.route('/search', methods=['POST'])
+@inject
+@logger.log_function_call()
+def get_filtered_news_endpoint(service: NewsService = Provide[Container.news_service]):
+    try:
+        filters = request.get_json() or {}
+        dto = SearchDTO().load(filters)
+        filtered_news = service.get_filtered_news(dto)
```

Comment:
When naming a variable the name must reflect whatever it is stored in that variable. If we had one service, then I'd justl let it slide, but since we have quite a lot of them, I'd suggest specifying which exact service it is, like: `news_service`. In this particular case it's a minor mistake, because we use service immediately after declaration, but usually you'd want a better name fort a variable